modules/cluster-autoscaler/main.tf

```diff
@@ -67,22 +76,14 @@ resource "aws_iam_role_policy" "cluster_autoscaler" {
   policy = data.aws_iam_policy_document.cluster_autoscaler.json
 }
 
-resource "kubernetes_namespace" "cluster_autoscaler" {
-  count = var.enable ? 1 : 0
-
-  metadata {
-    name = "cluster-autoscaler"
-  }
-}
-
 resource "helm_release" "cluster_autoscaler" {
   count = var.enable ? 1 : 0
 
-  name       = "aws-cluster-autoscaler"
-  chart      = "cluster-autoscaler"
-  version    = var.chart_version
-  repository = "https://kubernetes-charts.storage.googleapis.com"
-  namespace  = kubernetes_namespace.cluster_autoscaler.0.metadata.0.name
+  name       = local.release_name
+  chart      = local.chart_name
+  version    = local.chart_version
+  repository = local.repository
+  namespace  = local.namespace
```

Comment:
You need to set `create_namespace = true` in case it's overridden and non-existent.